common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -88,7 +88,7 @@ public MLToolSpec(StreamInput input) throws IOException {
         }
         this.tenantId = streamInputVersion.onOrAfter(VERSION_2_19_0) ? input.readOptionalString() : null;
         if (input.available() > 0 && input.readBoolean()) {
-            parameters = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
+            attributes = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
```

Comment:
is it the attributes and parameters are reading the same map? what is the intention?

        if (input.readBoolean()) {
            parameters = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
        }
        includeOutputInAgentResponse = input.readBoolean();
        if (input.getVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_FOR_TOOL_CONFIG) && input.readBoolean()) {
            configMap = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
        }
        this.tenantId = streamInputVersion.onOrAfter(VERSION_2_19_0) ? input.readOptionalString() : null;
        if (input.available() > 0 && input.readBoolean()) {
            attributes = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
        }